webapp/channels/src/components/copy_button.tsx

```diff
@@ -1,29 +1,53 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import type {Placement} from '@floating-ui/react';
 import classNames from 'classnames';
 import React, {useRef, useState} from 'react';
-import {Tooltip} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-import OverlayTrigger from 'components/overlay_trigger';
+import useTooltip from 'components/common/hooks/useTooltip';
 
-import Constants from 'utils/constants';
 import {t} from 'utils/i18n';
 import {copyToClipboard} from 'utils/utils';
 
 type Props = {
     content: string;
     beforeCopyText?: string;
     afterCopyText?: string;
-    placement?: string;
+    placement?: Placement;
     className?: string;
 };
 
 const CopyButton: React.FC<Props> = (props: Props) => {
     const [isCopied, setIsCopied] = useState(false);
     const timerRef = useRef<NodeJS.Timeout | null>(null);
 
+    const getDefaultMessage = () => {
+        if (isCopied) {
+            return props.afterCopyText;
+        }
+        return props.beforeCopyText ?? 'Copy code';
+    };
+
+    const getId = () => {
+        if (isCopied) {
+            return t('copied.message');
+        }
+        return props.beforeCopyText ? t('copy.text.message') : t('copy.code.message');
```

Comment:
you can `useIntl` here instead of t function

Reply:
Made the changes you asked for! 

Sorry for the bit of a delay, got caught up with some stuff. Hope that's cool.